tests/test_client_functional.py

```diff
@@ -2927,6 +2927,73 @@ async def test_creds_in_auth_and_url() -> None:
         await session.close()
 
 
+@pytest.mark.parametrize(
+    ["check_redirect_auth_precedence"],
+    ([False], [True]),
+    ids=("redirect auth clash allowed", "redirect auth precedence"),
```

Comment:
For a single param, you don't need to nest names or values. Additionally, it's best to use kebab-case in test IDs since these are often copied and pasted in terminal to run just one of the tests. And it's better to make test IDs short as they augment the test name, not replace it. It's not very clear what the variants are, I've taken a stub at trying to come up with new IDs:
```suggestion
    "check_redirect_auth_precedence",
    (False, True),
    ids=("clash-allowed", "precedence-enforced"),
```

Reply:
It appears you do need nesting even for a single param. If I don't nest I get

```
ERROR tests/test_client_functional.py - TypeError: object of type 'bool' has no len()
```

The rest is fixed as suggested :)

Reply:
No, nesting is not necessary. You should've accepted the patch from the GH UI. I've suggested the correct change that updates several lines and your error implies that you only copied it partially. In general, it's recommended to commit code suggestions via GH and then pull them locally, if necessary.

Reply:
Ugh, my bad. Fixed.